### PR TITLE
[main][feat] Upgrade executor

### PR DESCRIPTION
**Pull request by alpanaca**
## Description
- Add option to skip exposure check
- Add slippage control from manager

### CODE REVIEW

File: test/pcs-v3-executor-01/PCSV3Executor01_Swap.t.sol [hunk 1]
```diff
@@ -109,6 +109,21 @@ contract PCSV3Executor01SwapForkTest is BscFixture {
     avManager.manage(L_USDTBNB_05_PCS1, manageBytes);
   }
 
+    function testRevert_WhenSkipExposureCheck_WhenExposureBecomeMorePositiveAfterSwap_ShouldWork() external {
```
Comment (jr-alpaca) [inline test/pcs-v3-executor-01/PCSV3Executor01_Swap.t.sol:112]:
```suggestion
    function testCorrectness_WhenSkipExposureCheck_WhenExposureBecomeMorePositiveAfterSwap_ShouldWork() external {
```

File: test/pcs-v3-executor-01/PCSV3Executor01_Swap.t.sol [hunk 2]
```diff
@@ -122,6 +137,21 @@ contract PCSV3Executor01SwapForkTest is BscFixture {
     avManager.manage(L_USDTBNB_05_PCS1, manageBytes);
   }
 
+    function testRevert_WhenSkipExposureCheck_WhenExposureWentFromPositiveToNegative_ShouldWork() external {
```
Comment (jr-alpaca) [inline test/pcs-v3-executor-01/PCSV3Executor01_Swap.t.sol:140]:
```suggestion
    function testCorrectness_WhenSkipExposureCheck_WhenExposureWentFromPositiveToNegative_ShouldWork() external {
```

File: test/pcs-v3-executor-01/PCSV3Executor01_Swap.t.sol [hunk 3]
```diff
@@ -109,6 +109,21 @@ contract PCSV3Executor01SwapForkTest is BscFixture {
...
+    int256 exposure = oracle.getExposure(L_USDTBNB_05_PCS1, L_USDTBNB_05_PCS1_WORKER);
+    assertGt(exposure, 0);
+    bytes[] memory manageBytes = new bytes[](1);
+    manageBytes[0] = abi.encodeCall(PCSV3Executor01.swap, (address(wbnb), 2 ether, false));
+
+    vm.prank(executor.owner());
+    executor.setSkipExposureChecks(L_USDTBNB_05_PCS1, true);
+
+    vm.prank(manager);
+    avManager.manage(L_USDTBNB_05_PCS1, manageBytes);
```
Comment (jr-alpaca) [inline test/pcs-v3-executor-01/PCSV3Executor01_Swap.t.sol:124]: please add another assert to check if exposure actually increase.

File: src/executors/PCSV3Executor01.sol [hunk 4]
```diff
@@ -596,4 +618,9 @@ contract PCSV3Executor01 is Executor {
     vaultOracle = PancakeV3VaultOracle(_vaultOracle);
     emit LogSetVaultOracle(_vaultOracle);
   }
+
+  function setSkipExposureChecks(address _vaultToken, bool _skip) external onlyOwner {
+    skipExposureChecks[_vaultToken] = _skip;
```
[the remark this answers is not in the document]
Comment (alpanaca) [reply src/executors/PCSV3Executor01.sol:623]: Good to have but not necessary. If vaultToken doesn't exist, it would fail during manage anyway.

**Review by jr-alpaca — APPROVED** [review]

**Review by wow-alpaca — APPROVED** [review]

**Pull request merged**